src/anyp/Uri.cc

```diff
@@ -438,24 +419,55 @@ AnyP::Uri::parse(const HttpRequestMethod& method, const char *url)
         }
     }
 
-    parseFinish(protocol, proto, urlpath, foundHost, SBuf(login), foundPort);
+    setScheme(scheme);
+    path(urlpath);
+    host(foundHost);
+    userInfo(SBuf(login));
+    port(foundPort);
     return true;
+
+    } catch (...) {
+        debugs(23, 2, "URI parse error: " << CurrentException);
```

Comment:
The debug line will automatically say `Uri` and `parse` so we can simplify and reduce debugging noise:

```suggestion
        debugs(23, 2, "error: " << CurrentException);
```

BTW, reporting the original rawUrl (using the Raw class) may be a good idea here.